src/main/java/org/testobject/appium/junit/TestParser.java

```diff
@@ -8,39 +8,14 @@
 
 class TestParser {
 
-	private static final Pattern METHOD_AND_CLASS_NAME_PATTERN_STRICT = Pattern.compile("(.+)[\\[\\(](.*)[\\]\\)]");
-	private static final Pattern METHOD_AND_CLASS_NAME_PATTERN_LOOSE = Pattern.compile("(.+)");
-
-
 	public static Test from(Description testDescription) {
+		String className = testDescription.getClassName();
 
-		String className;
-		String methodName;
-		String deviceId = null;
-
-		Matcher matcher = METHOD_AND_CLASS_NAME_PATTERN_STRICT.matcher(testDescription.getMethodName());
```

Comment:
We still parse the test, we just now make sure the test is in a much simpler format (everything separated by spaces). 

I think the old parsing was very complicated, so the change makes sense.